src/core/compact_object.h

```diff
@@ -376,6 +377,10 @@ class CompactObj {
   // Returns the external data of the object incuding its ColdRecord.
   CoolItem GetCool() const;
 
+  // Prequisite: IsCool() is true.
+  // Keeps cool record only as external value and discard in-memory part.
+  void KeepExternal(size_t offset, size_t sz);
```

Comment:
`KeepExternal` is not clear imho.

`Freeze` - a more creative option, and maybe `DropCool` is more direct explaining what it does?